photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+        
+        if positions is None:
+            return self.do_photometry(image)
+        else:
+            return self.do_fixed_photometry(image, positions)
+
+    def do_photometry(self, image):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Image to perform photometry
+        
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        outtab = Table([[], [], [], [], [], []],
+                       names=('id', 'group_id', 'x_fit', 'y_fit', 'flux_fit',
+                              'iter_detected'),
+                       dtype=('i4', 'i4', 'f8', 'f8', 'f8', 'i4'))
+
+        residual_image = image.copy()
+        residual_image = residual_image - self.bkg(image)
+        sources = self.find(residual_image)
+        
+        if self.aperture_radius is None:
+            self.aperture_radius = self.psf.sigma.value*gaussian_sigma_to_fwhm
+
+        apertures = CircularAperture((sources['xcentroid'],
+                                      sources['ycentroid']),
+                                     r=self.aperture_radius)
+
+        sources['flux'] = aperture_photometry(residual_image,
+                                              apertures)['aperture_sum']
+        n = 1
+        while(n <= self.niters and len(sources) > 0):
+            intab = Table(names=['id', 'x_0', 'y_0', 'flux_0'],
+                          data=[sources['id'], sources['xcentroid'],
+                          sources['ycentroid'], sources['flux']])
+            star_groups = self.group(intab)
+            tab, residual_image = self.nstar(residual_image, star_groups)
+            tab['iter_detected'] = n*np.ones(tab['x_fit'].shape, dtype=np.int)
+            outtab = vstack([outtab, tab])
+            sources = self.find(residual_image)
+
+            if len(sources) > 0:
+                apertures = CircularAperture((sources['xcentroid'],
+                                              sources['ycentroid']),
+                                             r=self.aperture_radius)
+                sources['flux'] = aperture_photometry(residual_image,
+                                                      apertures)['aperture_sum']
+            n += 1
+        return outtab, residual_image
+
+    def do_fixed_photometry(self, image, positions):
+        """
```

Comment:
This needs an intro/description.  I.e., what is "fixed" meaning in this context?